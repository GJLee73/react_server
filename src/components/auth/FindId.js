import React from 'react';

class FindId extends React.Component {
	constructor(props) {
		super(props);
		this.state = { email:'', name:'' }
        this.handleChange=this.handleChange.bind(this)
        this.handleKeyPress=this.handleKeyPress.bind(this)
	}

	handleChange(e){ 
		let nextState = {};
		nextState[e.target.name] = e.target.value;
		this.setState(nextState);
	}
    handleKeyPress(e){
		if(e.charCode===13){
			
            this.props.sendEmail(this.state.name,this.state.email);
            
		
        }
    }
	render() { 
		return(
			<div>
				<div className="card-content">
					<div className="row">
						<div className="input-field col s12 username">
						<label>NAME</label>
						<input
						name="name"
						type="text"
						className="validate"
						onChange={this.handleChange}
						value={this.state.name}/>
						</div>
						<div className="input-field col s12">
							<label>email</label>
							<input
							name="email"
							type="text"
							className="validate"
							onChange={this.handleChange}
							value={this.state.email}
							onKeyPress={this.handleKeyPress}/>
						</div>
					<a className="waves-effect waves-light btn" onClick={()=>{this.props.sendEmail(this.state.name,this.state.email)}}>SUBMIT</a>
					</div>
				</div>
			</div>
		)
	}
}
 
export default FindId;