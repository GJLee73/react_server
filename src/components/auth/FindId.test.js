import React from 'react';
import { shallow} from 'enzyme';
import FindId from './FindId'
 

describe('findId test',()=>{

	it('renders without crashing', () => {
		const findId=shallow(<FindId />);
		expect(findId).toMatchSnapshot()
	  });
})
