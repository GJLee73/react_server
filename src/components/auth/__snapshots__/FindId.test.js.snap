// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`findId test renders without crashing 1`] = `
<div>
  <div
    className="card-content"
  >
    <div
      className="row"
    >
      <div
        className="input-field col s12 username"
      >
        <label>
          NAME
        </label>
        <input
          className="validate"
          name="name"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="input-field col s12"
      >
        <label>
          email
        </label>
        <input
          className="validate"
          name="email"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <a
        className="waves-effect waves-light btn"
        onClick={[Function]}
      >
        SUBMIT
      </a>
    </div>
  </div>
</div>
`;
